lib/game_team.rb

```diff
@@ -0,0 +1,48 @@
+class GameTeam
+#
+  @@all = nil
+
+  def self.all
+    @@all
+  end
+
+  def self.from_csv(csv_file_path)
+    csv = CSV.read("#{csv_file_path}", headers: true, header_converters: :symbol)
+    @@all = csv.map { |row| GameTeam.new(row) }
+  end
+
+    attr_reader :game_id,
+                :team_id,
+                :hoa,
+                :result,
+                :settled_in,
+                :head_coach,
+                :goals,
+                :shots,
+                :tackles,
+                :pim,
+                :power_play_opportunities,
+                :power_play_goals,
+                :face_off_win_percentage,
+                :giveaways,
+                :takeaways
+
+  def initialize(details)
+    @game_id = details[:game_id].to_i
+    @team_id = details[:team_id].to_i
+    @hoa = details[:hoa]
+    @result = details[:result]
+    @settled_in = details[:settled_in]
+    @head_coach = details[:head_coach]
+    @goals = details[:goals].to_i
+    @shots = details[:shots].to_i
+    @tackles = details[:tackles].to_i
+    @pim = details[:pim].to_i
```

Comment:
note that the headers are camel case, then when pulled in they are downcased. Ruby convention dictates snake case